src/cider/nrepl/middleware/debug.clj

```diff
@@ -28,18 +28,22 @@
   "Send value and coordinates to the client through the debug channel.
   Sends a response to the message stored in debugger-message."
   [value extras]
-  ;; Notify cider that a (break) is incoming, along with the value of
-  ;; the instrumented sexp and instructions on how to find it in the
-  ;; code.
-  (transport/send
-   (:transport @debugger-message)
-   (response-for @debugger-message
-                 (assoc extras
-                        :debug-value (pr-str value)
-                        :breakfunction nil)))
-  ;; Send the actual break.
-  (binding [*out* (new java.io.StringWriter)]
-    (break value)))
+  ;; Don't debug code being run synchronously. It is probably
+  ;; unintended and will unexpectedly ask for input.
+  (if (:sync clojure.tools.nrepl.middleware.interruptible-eval/*msg*)
```

Comment:
Ah, I think I get your question now. I suppose there's nothing preventing other middleware from `eval`ing code as well.
In fact, I think pprint might be one.